src/ast.rs

```diff
@@ -0,0 +1,20 @@
+#[derive(Debug, PartialEq, Clone)]
+pub enum Literal {
+    // primitiveLiteral
+    Null,
+    Boolean(bool),
+    Date(String),
+    DateTimeOffset(String),
+    Time(String),
```

Comment:
I'm not yet sure whether I should use native types for date/time related types, as I don't think they easily map onto Python date/time types.